src/network/client_system.cpp

```diff
@@ -171,20 +155,118 @@ void ClientSystem::Receive( ENetEvent& event )
     //         event.channelID);
     std::istringstream iss( std::string( ( char* )( event.packet->data ), event.packet->dataLength ) );
     eos::portable_iarchive ia( iss );
-    if ( mMessageHolder.GetIncomingMessages().mMessages.empty() )
+
+    MessageList::Messages_t messages;
+    ia >> messages;
+    mMessageHolder.GetIncomingMessages().TransferFrom( messages );
+
+    /* Clean up the packet now that we're done using it. */
+    enet_packet_destroy ( event.packet );
+}
+
+void ClientSystem::UpdateThread()
+{
+    while (mRunning)
```

Comment:
Two things can trigger:
- Spurious wakeup call. That doesn't happen too often, and we don't mind to run the pull on network.
- notify trigger from the main thread. That's the time when we surely want to send stuff. This can happen 3 times per a frame. 180 times per a second. It is not a performance issue at all i think the poll is pretty cheap. considering that the mWaitMillisecs is set to 5 from settings now. But will see. The problem is in the return statement :dagger: